### PR TITLE
CVE-2015-6777 and CVE-2013-6636

### PR DESCRIPTION
**Pull request by mbillow**

### CODE REVIEW

File: cves/CVE-2013-6636.yml [hunk 1]
```diff
@@ -15,7 +15,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
```
Comment (andymeneely) [inline cves/CVE-2013-6636.yml]: Need CWE

File: cves/CVE-2013-6636.yml [hunk 2]
```diff
@@ -15,7 +15,12 @@ description_instructions: |
...
   keep too.
-description: 
+description: |
+  The FrameLoader::notifyIfInitialDocumentAccessed function in
```
Comment (andymeneely) [inline cves/CVE-2013-6636.yml]: Can't be jargony - don't just take it from the CVE.

File: cves/CVE-2013-6636.yml [hunk 3]
```diff
@@ -75,32 +85,30 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: Based on the Bug Report and files modified.
+  name: Blink
```
Comment (andymeneely) [inline cves/CVE-2013-6636.yml]: Can you be more specific thank blink? That's a big part of the system.

File: cves/CVE-2013-6636.yml [hunk 4]
```diff
@@ -75,32 +85,30 @@ discovered:
...
 
     Write a brief (under 100 words) description of why you think this commit was
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
-  commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  commits: []
```
Comment (andymeneely) [inline cves/CVE-2013-6636.yml]: I need this part to be completed - show me that you actually researched this.

File: cves/CVE-2013-6636.yml [hunk 5]
```diff
@@ -175,4 +183,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
```
Comment (andymeneely) [inline cves/CVE-2013-6636.yml]: I need much more analysis. Do more on the interesting commits and look at what the project was doing. Incorporate the entire SE process into this, and to do that, you'll need to understand _their_ process. Just more context. Imagine you're presenting this to a CEO and you're trying to explain why something happened so we can improve things.

File: cves/CVE-2015-6777.yml [hunk 6]
```diff
@@ -15,7 +15,12 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Use-after-free vulnerability in the ContainerNode::notifyNodeInsertedInternal
```
Comment (andymeneely) [inline cves/CVE-2015-6777.yml]: Same as above - no jargon. See the directions. Check out some of the "Exemplary Work" PRs - see how it's a useful description for the uninitiated.

File: cves/CVE-2015-6777.yml [hunk 7]
```diff
@@ -172,4 +179,6 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This error was definitely a coding mistake. The bug was found in an error
```
Comment (andymeneely) [inline cves/CVE-2015-6777.yml]: Yeah not enough. This sentence could be true about literally every single bug in every software project ever.

**Comment by mbillow** [comment]
@andymeneely I went through and added more to most of my responses and rewrote a couple things to be more clear and have less jargon. Let me know if you think anything else should be modified or if any answer could use more information. Thanks!

File: cves/CVE-2015-6777.yml [hunk 8]
```diff
@@ -180,5 +183,8 @@ mistakes:
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
   answer: |
-    This error was definitely a coding mistake. The bug was found in an error
-    report and then fixed. 
+    This error was definitely a very simple coding mistake. The original commit 
+    that intorduced this error was meant to optimize node indertion by not
```
Comment (qwertyflagstop) [inline cves/CVE-2015-6777.yml]: I think you mean node iteration not inderation.

File: cves/CVE-2013-6636.yml [hunk 9]
```diff
@@ -143,8 +151,8 @@ lessons:
     applies: 
     note: 
   distrust_input:
-    applies: 
-    note: 
+    applies: true
+    note: User input was the cause of this issue. Additional checked were needed.
```
Comment (qwertyflagstop) [inline cves/CVE-2013-6636.yml]: How so? In the description you say there is an incorrect check for an empty document during presentation 
of a modal dialog. Can a user cause this event to happen with malicious input?

File: cves/CVE-2015-6777.yml [hunk 10]
```diff
@@ -72,20 +81,20 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: The portion of the repo the vulnerability was found in is WebKit.
+  name: WebKit
```
Comment (qwertyflagstop) [inline cves/CVE-2015-6777.yml]: Webkit is fairly large. Is it possible to be more specific?

File: cves/CVE-2015-6777.yml (hunk 8 above)
Comment (mbillow) [inline cves/CVE-2015-6777.yml]: Actually, it was supposed to be `insertion`. Thanks.

File: cves/CVE-2015-6777.yml (hunk 10 above)
Comment (mbillow) [inline cves/CVE-2015-6777.yml]: It is actually a part of the core DOM generation functionality in WebKit.

File: cves/CVE-2013-6636.yml [hunk 11]
```diff
@@ -15,7 +15,16 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Versions of Google Chrome before 31.0.1650.63 made an incorrect check for 
+  an empty document during presentation of a modal dialog, which allows 
+  remote attackers to spoof the address bar via vectors involving the 
+  document.write method. These vectors involved calling document.write on a 
+  window opened by window.open and then Chrome not correctly setting the URL 
+  in the OmniBox.
```
Comment (hqp6684) [inline cves/CVE-2013-6636.yml]: A brief definition of OminiBox would be nice

File: cves/CVE-2015-6777.yml [hunk 12]
```diff
@@ -15,7 +15,15 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  In versions of Google Chrome before 47.0.2526.73 there was a vulnerabilty that
+  would cause the application to crash due to memory that was referenced after
+  being freed py the application. This problem could have been leveraged to cause
```
Comment (hqp6684) [inline cves/CVE-2015-6777.yml]: Spelling, py => by ?

**Pull request merged**